apps/native-tests/ios/Podfile.properties.json

```diff
@@ -1,3 +1,3 @@
 {
-  "expo.jsEngine": "jsc"
+  "expo.jsEngine": "hermes"
```

Comment:
Any specific reason why we changed this? As far as I remember we don't use JS at all in this app

Reply:
The Shared object tests were failing on jsc. I bumped the Xcode version and switched to Hermes and it worked a lot better. I think we went back to Xcode 14 so maybe this needs to be changed but might have the same issue with the tests.

Reply:
Imho we should switch it back to JSC, for these reasons:
- it wouldn't need to download and link hermes (faster build, but probably not that much since it's no longer built from sources)
- we don't test JSC at all since we added support for Hermes, so I intentionally used JSC here to at least have some small coverage 😅 

I know these tests started failing at some point, but I think we should fix that issue instead of changing the engine.